tests/Validator/UniqueEnumValueNamesTest.php

```diff
@@ -0,0 +1,274 @@
+                    'locations' => [
+                        ['line' => 4, 'column' => 9],
+                        ['line' => 6, 'column' => 9],
+                    ],
+                ],
+            ],
+        );
+    }
+
+    /**
+     * @see it('duplicate value inside different extensions')
+     */
+    public function testDuplicateValueInsideDifferentExtensions(): void
+    {
+        $this->expectSDLErrorsFromRule(
+            new UniqueEnumValueNames(),
+            '
+      enum SomeEnum
+      extend enum SomeEnum {
+        FOO
+      }
+      extend enum SomeEnum {
+        FOO
+      }
+        ',
+            null,
+            [
+                [
+                    'message' => 'Enum value "SomeEnum.FOO" can only be defined once.',
+                    'locations' => [
+                        ['line' => 4, 'column' => 9],
+                        ['line' => 7, 'column' => 9],
+                    ],
+                ],
+            ],
+        );
+    }
+
+    /**
+     * @see it('adding new value to the type inside existing schema')
+     */
+    public function testAddingNewValueToTheTypeInsideExistingSchema(): void
+    {
+        $this->expectValidSDL(
+            new UniqueEnumValueNames(),
+            '
+      extend enum SomeEnum {
+        FOO
+      }
+        ',
+            BuildSchema::build('enum SomeEnum'),
+        );
+    }
+
+    /**
+     * @see it('adding conflicting value to existing schema twice')
+     */
+    public function testAddingConflictingValueToExistingSchemaTwice(): void
+    {
+        $this->expectSDLErrorsFromRule(
+            new UniqueEnumValueNames(),
+            '
+      extend enum SomeEnum {
+        FOO
+      }
+      extend enum SomeEnum {
+        FOO
+      }
+        ',
+            BuildSchema::build('
+      enum SomeEnum {
+        FOO
+      }
+            '),
+            [
+                [
+                    'message' => 'Enum value "SomeEnum.FOO" already exists in the schema. It cannot also be defined in this type extension.',
+                    'locations' => [
+                        ['line' => 3, 'column' => 9],
+                    ],
+                ],
+                [
+                    'message' => 'Enum value "SomeEnum.FOO" already exists in the schema. It cannot also be defined in this type extension.',
+                    'locations' => [
+                        ['line' => 6, 'column' => 9],
+                    ],
+                ],
+            ],
+        );
+    }
+
+    /**
+     * @see it('adding enum values to existing schema twice')
+     */
+    public function testAddingEnumValuesToExistingSchemaTwice(): void
+    {
+        $this->expectSDLErrorsFromRule(
+            new UniqueEnumValueNames(),
+            '
+      extend enum SomeEnum {
+        FOO
+      }
+      extend enum SomeEnum {
+        FOO
+      }
+        ',
+            BuildSchema::build('
```

Comment:
I would prefer to build the schema before calling expect and pass `$schema` here. That way the test structure would be more similar to `graphql-js` and the code easier to compare. 

Applies also to the tests above.